File: week-01/ek_6198_garden.py
import sys
read = sys.stdin.readline

n = int(read().rstrip())

st = []
ans = 0
for _ in range(n):
    temp = int(read().rstrip())

    while st and st[-1] <= temp:
        st.pop()

    ans += len(st)
    st.append(temp)

print(ans)

File: week-01/ek_6198_garden.cpp
#include <iostream>
#include <stack>
using namespace std;

int main(void)
{
	ios_base::sync_with_stdio(false);
	cin.tie(nullptr);

	int n, temp;
	cin >> n;

	stack<int> st;
	long long ans = 0;

	while (n--)
	{
		cin >> temp;

		while (!st.empty() && st.top() <= temp)
			st.pop();

		ans += st.size();
		st.push(temp);
	}

	cout << ans;

	return 0;
}